src/upload-lib.ts

```
@@ -177,6 +177,9 @@ function getSarifFilePaths(sarifPath: string) {
 export function countResultsInSarif(sarif: string): number {
   let numResults = 0;
   for (const run of JSON.parse(sarif).runs) {
```

Shall we also handle the case where `runs` is null/undefined? It's valid SARIF according to https://docs.oasis-open.org/sarif/sarif/v2.1.0/csprd01/sarif-v2.1.0-csprd01.html#_Toc10540920 but I think we currently crash rather than returning 0.

---

Sure. Probably a good idea to be more defensive in general. I'll take a look and see if there are any more simple changes to make that will strengthen our handling of invalid sarif.